### ISSUE TITLE
Move from <title> to aria-label

### ISSUE THREAD

**Issue by birjolaxew** [simple-icons/simple-icons#1191]
As discussed in #1189, using `<title>` has the unfortunate side-effect of displaying a tooltip on hover in most browsers. As this usually isn't what we want, we should switch to using the `aria-label` attribute instead.

As [put by MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute):

> The `aria-label` attribute is used to define a string that labels the current element. Use it in cases where a text label is not visible on the screen. If there is visible text labeling the element, use `aria-labelledby` instead.

**Comment by ericcornelissen** [comment]
This has been discussed in #1135 before, and I would love to make this switch as well. However, I explained my reasoning for not doing so in [this comment](https://github.com/simple-icons/simple-icons/issues/1135#issuecomment-454674687). If my reasoning there is flawed, please point it out so we can make this change 😃

**Comment by davidklebanoff** [comment]
@birjolaxew - I agree that the tooltips are an unwelcome side effect and @ericcornelissen summed up the conundrum pretty nicely in the linked comment.

While we wait for the SVG 2 spec (which may never pass, apparently it's not looking good) it may be possible to address the tooltip problem via some css.

One could do a global css rule like:
```css
svg {
  pointer-events: none;
}
```

or if we so chose, we could embed the css rule in every icon via a style attribute.

It would need extensive testing as disabling pointer events on an element is a bit of a sledgehammer type approach that could have unintended side effects.

**Comment by ericcornelissen** [comment]
I don't think this is a good idea, @davidklebanoff, because of the side effects.

For example, I think a pretty common usage for the icons we provide is to use it as a button that links to a site (e.g. the Facebook or Twitter logo to link to your own Facebook or Twitter profile). Adding `pointer-events: none;` would also disable click events and as a result these links may no longer work (depending on how the HTML is structured).

Besides, embedding it in the icon leads to a lot of redundancy since most users probably use multiple icons.

That being said, we could add it as a note in the documentation but with a disclaimer about the side effects. But I'm personally not a fan of this since the benefit is very minor.

**Comment by davidklebanoff** [comment]
I guess my vote then is to put a mention of the code snippet and its impact in the documentation, and people can use it at their own risk. I have come across the tooltip problem in the past, so I understand it's potential to be a pain point for some.

I know you qualified your answer with "depending on how the html is structured". Thinking on it a bit, I don't think it will disable any click events in any common use cases. The pointer events would be disabled on the SVG itself, so any wrapping element should be fine (i.e. link, button).

I believe it would only impact a click event on the icon itself. So it's impact is probably limited to manually added javascript events? If someone feels comfortable enough adding javascript events to their website, then hopefully they'll also understand the impact of adding a `pointer-events: none` to their css.

Playground:
https://jsbin.com/kuyihuyodi/edit?html,output

**Comment by ericcornelissen** [comment]
> I know you qualified your answer with "depending on how the html is structured". Thinking on it a bit, I don't think it will disable any click events in any common use cases. The pointer events would be disabled on the SVG itself, so any wrapping element should be fine (i.e. link, button).

That's fair and probably one of the more common use cases, still...

> I believe it would only impact a click event on the icon itself. So it's impact is probably limited to manually added javascript events? 

That is indeed the case (I should've been more clear on that in my original comment) see [this jsbin](https://jsbin.com/kofadegemu/1/edit?console,output).

> If someone feels comfortable enough adding javascript events to their website, then hopefully they'll also understand the impact of adding a `pointer-events: none` to their css.

I don't agree 🤔 I personally wouldn't assume that. I don't think our users are stupid, but I feel more comfortable not assuming this kind of knowledge. Just consider the fact that we need to discuss and test it manually.

* * *

All in all, I definitely think we shouldn't embed it because of the redundancy. Especially given the length we go to reduce redundancy so far (e.g. requiring SVG optimization and debating a 2% filesize reduction in [this comment](https://github.com/simple-icons/simple-icons/issues/1135#issuecomment-449317147)).

But, adding it to the documentation somehow is not a bad idea 👍

**Comment by jorgeamadosoria** [comment]
@ericcornelissen why is this ticket still open? It seems that everything that was to be said has been said, specially in the other tickets.
Should we add to the documentation as suggested and close it?

**Comment by ericcornelissen** [comment]
It's still open because we want to change but "can't" yet (or at least not last time I checked) - even if we add documentation related to this I would leave it open. Why do you want to close it?

As for "the documentation", any suggestions?

**Comment by jorgeamadosoria** [comment]
> It's still open because we want to change but "can't" yet (or at least not last time I checked) - even if we add documentation related to this I would leave it open. Why do you want to close it?

I want to close as many tickets as possible. There are almost 400 tickets opened. That's way too much to handle.

> 
> As for "the documentation", any suggestions?

Just a line about "we are using title for accessibilty reasons" or something like that. It would qualify as a FAQ.
Incidentally, do we have a FAQ? Shouldn't we have a FAQ for precisely this kind of things?

**Comment by ericcornelissen** [comment]
> I want to close as many tickets as possible. There are almost 400 tickets opened. That's way too much to handle.

In that regard, I'd say we have labels to help out a bit. There's just 7 open tickets with the `package` label (or 22 with the label `meta`, which could be more appropriate here) and on top of that we use the `pending` label to indicate that an issue needs attention later. Also, it's not your job to close all tickets. Still, if you feel it helps you as a maintainer, feel free to close this - provided you have checked whether it makes sense to switch to `aria-label` now.


> Just a line about "we are using title for accessibilty reasons" or something like that. It would qualify as a FAQ.
Incidentally, do we have a FAQ? Shouldn't we have a FAQ for precisely this kind of things?

No FAQ yet. You can add it for "this kind of thing" (don't know what else you have in mind though). Regarding the wording, I'd be a bit more verbose, e.g.: "SVGs have a `<title>` node to provide a minimum level of accessibility. Further information is available in [issue #1191](https://github.com/simple-icons/simple-icons/issues/1191)."

**Comment by adamrusted** [comment]
Looks like `aria-label` has pretty universal usage these days.
@simple-icons/maintainers - how would we feel about adding this into our next major release?

**Comment by PratyushJayachandran** [comment]
> Looks like `aria-label` has [pretty universal usage](https://caniuse.com/mdn-api_element_arialabel) these days. @simple-icons/maintainers - how would we feel about adding this into our next major release?

I agree

**Comment by LitoMore** [comment]
I don't think so. We are just SVG resources provider. Our user can wrap/modify the HTML code as they want.

Most of our users are using `<img src="CDN_URL" />` to load icons. The `aria-label` inside our SVG has no effect in fact. They should add `alt` and our attributes to their `<img>` node.

**Comment by adamrusted** [comment]
> Most of our users are using `<img src="CDN_URL" />` to load icons. The `aria-label` inside our SVG has no effect in fact. They should add `alt` and other attributes to their `<img>` node.

If we go down that route, we should remove the title tag too.

I don't agree with _most_ users going via a CDN, in the ecosystem of third-party extensions we have, most are either using our NPM library or directly using our repo as a gitmodule. We should be setting an example by utilising best practice for accessibility where we can.

**Comment by LitoMore** [comment]
Even our npm module then won't use the SVG file directly. The only possibility they have of using our SVG is the `icon.svg`.

Then the code will be, it's not a recommended API:

```jsx
<div dangerouslySetInnerHTML={icon.svg} />
```

If they're using `<img>`, then they should always use the `icon.path` instead. And it's super easy to add `aria-label` or other attributes:

```jsx
<svg path={icon.path} />
```

On the other hand, if you agree with `aria-label`, more people will submit similar requests in the future. For example, the `class` attribute:

```jsx
<svg class="si-simpleicons" aria-label="Simple Icons" ...>
	<title>Simple Icons</title>
	<path d="..." />
</svg>
```

```css
.si-simpleicons {
	color: blue;
}
```

This looks cool and useful, but it's unnecessary for all users. They should implement this on their side rather than our library.

**Comment by adamrusted** [comment]
I don't think comparing `aria-label` to classes is fair. One is an accessibility feature we should be striving to include where we can, whereas the other is purely stylistic.

If we were to stagger the release of this, adding the `aria-label` attribute in one release, and deprecated `title` in the next, then yes - we'll have some additional file size for 1 major release, but it'd give those that _do_ rely on `<title>` to make the change.

**Comment by service-paradis** [comment]
> I don't think comparing aria-label to classes is fair. One is an accessibility feature we should be striving to include where we can, whereas the other is purely stylistic.

Totally agree with you on this. Classes and aria attributes are not really comparable.

> Most of our users are using <img src="CDN_URL" /> to load icons.

I would really love to know the statistics behind this affirmation! On a personal note, I always try to be CDN free since it relies on external reliability and add unnecessary requests on page load. Am i the only one?

> <div dangerouslySetInnerHTML={icon.svg} />

There are a lot of way other than this to use SVGs!

> Some features look cool and useful, but they may not be necessary for all users.

I dont consider aria attributes "cool features". Those are quite necessary nowadays.

> If we were to stagger the release of this, adding the aria-label attribute in one release, and deprecated title in the next, then yes - we'll have some additional file size for 1 major release, but it'd give those that do rely on <title> to make the change.

A compromise, that would increase sizes, would be to keep `title` combined with `aria-labelledby`. Ex.

```svg
<svg aria-labelledby="icon-title">
<title id="icon-title">Brand Name</title>
...
</svg>
```

**Comment by adamrusted** [comment]
> A compromise, that would increase sizes, would be to keep `title` combined with `aria-labelledby`. Ex.
> 
> ```
> <svg aria-labelledby="icon-title">
> <title id="icon-title">Brand Name</title>
> ...
> </svg>
> ```

Only thing to be aware of is if we use this approach, we'd need a unique ID for every icon, assuming users will use multiples in a project.

**Comment by PeterShaggyNoble** [comment]
Unfortunately, the point Eric made [here](https://github.com/simple-icons/simple-icons/issues/1135#issuecomment-454674687) is still valid so I don't think we could _default_ to `aria-label`. What I would be love to be able to do, though, is to offer people the option wherever possible of whether to use `title`, `aria-label` or neither by rewriting the SVG on the fly.

**Comment by adamrusted** [comment]
I think we can _somewhat_ bypass Eric's point. Although `aria-label` hasn't been accepted into the SVG spec itself, the attribute has universal availability in browsers, so would be parsed appropriately.

**Comment by mondeja** [comment]
> I think we can _somewhat_ bypass Eric's point. Although `aria-label` hasn't been accepted into the SVG spec itself, the attribute has universal availability in browsers, so would be parsed appropriately.

Note that we are not only distributing for Browsers, SVGs can be used in other contexts like desktop applications.

**Comment by PeterShaggyNoble** [comment]
I don't know if my last suggestion would work; a lot of people aren't familiar with SVG markup - I certainly wasn't until I joined MDI - so they're not going to know or care which one they need, if either. But, then again, a toggle might be a nice addition to our website for those who are familiar with SVG markup.
